File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/HorizontalPosition.java
/* Copyright Airship and Contributors */

package com.urbanairship.android.layout.property;

import android.view.Gravity;

import java.util.Locale;

import androidx.annotation.NonNull;

import com.urbanairship.json.JsonException;

public enum HorizontalPosition {
    START("start", Gravity.START),
    END("end", Gravity.END),
    CENTER("center", Gravity.CENTER_HORIZONTAL);

    @NonNull
    private final String value;
    private final int gravity;

    HorizontalPosition(@NonNull String value, int gravity) {
        this.value = value;
        this.gravity = gravity;
    }

    @NonNull
    public static HorizontalPosition from(@NonNull String value) throws JsonException {
        for (HorizontalPosition hp : HorizontalPosition.values()) {
            if (hp.value.equals(value.toLowerCase(Locale.ROOT))) {
                return hp;
            }
        }
        throw new JsonException("Unknown HorizontalPosition value: " + value);
    }

    public int getGravity() {
        return gravity;
    }

    @NonNull
    @Override
    public String toString() {
        return name().toLowerCase(Locale.ROOT);
    }
}

File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/HorizontalPosition.kt
/* Copyright Airship and Contributors */
package com.urbanairship.android.layout.property

import android.view.Gravity
import com.urbanairship.json.JsonException
import com.urbanairship.json.JsonValue

public enum class HorizontalPosition(
    private val value: String,
    @JvmField public val gravity: Int
) {

    START("start", Gravity.START),
    END("end", Gravity.END),
    CENTER("center", Gravity.CENTER_HORIZONTAL);

    override fun toString(): String {
        return name.lowercase()
    }

    public companion object {

        @JvmStatic
        @Throws(JsonException::class)
        public fun from(value: JsonValue): HorizontalPosition {
            val content = value.requireString().lowercase()
            return entries.firstOrNull { it.value == content }
                ?: throw JsonException("Unknown HorizontalPosition value: $value")
        }
    }
}
